packages/flutter/lib/src/painting/inline_span.dart

```diff
@@ -0,0 +1,257 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui show ParagraphBuilder;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/services.dart';
+
+import 'basic_types.dart';
+import 'text_style.dart';
+import 'text_painter.dart';
+
+/// An immutable span of inline content which form a paragraph.
+///
+/// The subclass [TextSpan] specifies text and may contain child [InlineSpan]s.
+///
+/// The subclass [WidgetSpan] specifies embedded inline widgets. Specify a
+/// widget by wrapping the widget with a [WidgetSpan]. The widget will be
+/// laid out inline within the paragraph.
+///
+/// Only [TextSpan] may contain children, so in order to include multiple
+/// [InlineSpan]s, the root node must be a [TextSpan]. Leaving the
+/// [TextSpan.text] field null results in the [TextSpan] acting as an empty
+/// node with a list of children.
+///
+/// {@tool sample}
+///
+/// This example shows a tree of [InlineSpan]s that make a query asking for a
+/// name with a [TextField] embedded inline.
+///
+/// ```dart
+/// RichText(
+///   text: TextSpan(
+///     text: 'My name is ',
+///     style: TextStyle(color: Colors.black),
+///     children: <InlineSpan>[
+///       WidgetSpan(
+///         alignment: InlineWidgetAlignment.baseline,
+///         baseline: TextBaseline.alphabetic,
+///         widget: ConstrainedBox(
+///           constraints: BoxConstraints(maxWidth: 100),
+///           child: TextField,
```

Comment:
does this code compile? i'm surprised the analyzer didn't complain about this?

Reply:
if the analyzer didn't complain about this please track down why

Reply:
(by "the analyzer" i mean the precommit tests -- they're supposed to analyze the code snippets in dartdocs)

Reply:
The analyzer hasn't run over this yet, as the engine patch this is dependent on hasn't been landed.

I will do a final pass on formatting and whatnot after the engine lands. I didn't want to land the engine too early just in case there were relevant changes to make on the engine side as a result of this patch.